Jellyfin.Api/Controllers/DevicesController.cs

```diff
@@ -111,28 +114,34 @@ public async Task<ActionResult> UpdateDeviceOptions(
     }
 
     /// <summary>
-    /// Deletes a device.
+    /// Deletes devices.
     /// </summary>
-    /// <param name="id">Device Id.</param>
+    /// <param name="id">Optional. Device Id.</param>
+    /// <param name="ids">Device Ids.</param>
     /// <response code="204">Device deleted.</response>
     /// <response code="404">Device not found.</response>
-    /// <returns>A <see cref="NoContentResult"/> on success, or a <see cref="NotFoundResult"/> if the device could not be found.</returns>
+    /// <returns>A <see cref="NoContentResult"/> on success, or a <see cref="NotFoundResult"/> if a device could not be found.</returns>
     [HttpDelete]
     [ProducesResponseType(StatusCodes.Status204NoContent)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<ActionResult> DeleteDevice([FromQuery, Required] string id)
+    public async Task<ActionResult> DeleteDevice([FromQuery, ParameterObsolete] string? id, [FromQuery, ModelBinder(typeof(CommaDelimitedArrayModelBinder))] string[] ids)
     {
-        var existingDevice = _deviceManager.GetDevice(id);
-        if (existingDevice is null)
+        string[] allIDs = ids.Concat(id != null ? [id] : Array.Empty<string>()).ToArray();
+
+        foreach (var i in allIDs)
```

Comment:
```suggestion
        foreach (var id in allIDs)
```

There is no rule that you must use `i` for variable names in a for(each) loop

Reply:
i for index in for loops is acceptable but not in foreach where the variable should be a self explanetory name